### DISCUSSION TITLE
Modification of DynamicGenerator

### DISCUSSION THREAD

**Discussion by mkaracelebi** [NREL-SIIP/PowerSimulationsDynamics.jl#184] [Q&A]
Hello everyone,

I am trying to observe the impact of different controllers in a synchronous machine. What might be the easiest way to change the AVR of a machine? Should I generate a new DynamicGenerator component from scratch or can I simply change the AVR? 

Many thanks for the support.

**Comment by jd-lara** [comment]
Sorry for the late reply, I just saw this. You need to change all the generators not just the AVR since the indexing in the data model is not automatically updated. The only way is calling [`set_dynamic_injector!()`](https://nrel-siip.github.io/PowerSystems.jl/stable/api/public/#PowerSystems.set_dynamic_injector!-Union{Tuple{U},%20Tuple{T},%20Tuple{T,%20U}}%20where%20{T%3C:StaticInjection,%20U%3C:Union{Nothing,%20DynamicInjection}})  with new generators every time.
